In this research of the **riveting** titanic database, I am looking for someone similar to my Grandma. My grandma is 77, around 2nd class and is female (obviously). I think of my grandma as a big spender as I always bring her Amazon packages when I visit, so I will be looking for a higher fare too.

**Critera:**
* Name
* Age: 70-77
* Pclass: 2
* Sex: female
* Fare: 100 or higher


In [4]:
import pandas as pd
import sqlite3

conn = sqlite3.connect('titanic.db')
print("Database connected. I'm ready, Freddy!")

Database connected. I'm ready, Freddy!


In [9]:
query1 = """
SELECT Name, Age, Pclass, Sex, Fare
FROM Passengers
WHERE Age <= 77 AND Age >= 70 AND Pclass = 2 AND Sex = 'female' AND Fare >= 100
ORDER BY Age DESC
"""

result = pd.read_sql(query1, conn)
display(result)


Empty DataFrame
Columns: [name, age, pclass, sex, fare]
Index: []

I did not find anybody close to my Grandma with this critera, so I will now adjust it to be less specific.

**What I will Remove:**
* Fare
* Pclass

**New Critera:**
* Name
* Age
* Sex
  



In [11]:
query2 = """
SELECT Name, Age, Pclass, Sex, Fare
FROM Passengers
WHERE Age <= 77 AND Age >= 70 AND Sex = 'female'
ORDER BY Age DESC
"""

result = pd.read_sql(query2, conn)
display(result)

name   age  pclass     sex  \
0  Cavendish, Mrs. Tyrell William (Julia Florence...  76.0       1  female   

    fare  
0  78.85

Finally! We found one! Her name is Mrs. Tyrell William Cavendish. Her age at the time was one year younger than my grandma, she is first class (an upgrade from what I was looking for), and payed a fare of 78.85. After adjusting my critera, I was pretty surprised that I found someone this close. I'm pretty happy with my result!

In [14]:
query3 = """
SELECT Name, Survived
FROM Passengers
WHERE Age <= 77 AND Age >= 70 AND Sex = 'female' 
ORDER BY Age DESC
"""

result = pd.read_sql(query3, conn)
display(result)

name  survived
0  Cavendish, Mrs. Tyrell William (Julia Florence...         1

I did one last query to see if she survived and luckily she did! 

**Report on the person I found**
   
    I found a person named Mrs. Tyrell William Cavendish, also known as Julia Florence Cavendish, through my research above. I specifically looked for someone that was in similar age to my grandma, which led me to filter for women between the ages of 70-77. Since my grandma is 77, finding Julia, who was 76 was almost a perfect match.
   
    My research on Julia Florence Cavendish showed that she was actually a 25-year-old woman at the time of the disaster, though she died much later at the age of 76. She was traveling with her husband, Tyrell, and her maid, Ellen Mary Barber, to visit her father in New York. Julia was rescued in lifeboat 6, but unfortunately, her husband was lost in the sinking. In a moving account, she recalled her last moments with him, saying, "Just as the lifeboat was lowered I again kissed my husband". He had assured her he would be safe and urged her to stay in the boat while the bright skies illuminated the scene. Learning these specific details made the database entry about her fare and class feel much more significant and heartbreaking. It really puts into perspective how quickly life changed for the survivors that night.
    
    One of the most surprising things I discovered was the dramatic life Julia led after the Titanic sank. She ended up facing a massive lawsuit over her home, Crakemarsh Hall, filed by a one-legged cab driver who claimed he was the rightful heir to her husband's estate. Around the same time, her father’s massive department store empire in the USA began to crumble into bankruptcy and legal scandal. It was interesting to see that while the database listed her fare as 78.85, the historical records show her father was known as the "Merchant Prince" and was incredibly wealthy before his downfall. I also realized that the "76" in my database was actually her age when she died in 1963, not her age during the voyage. This connection between the database age and her death certificate was a neat discovery that cleared up my initial confusion. She never remarried and remained by herself for the rest of her life.

**Works Cited:**
"Julia Florence Cavendish." Encyclopedia Titanica, 2020. https://www.encyclopedia-titanica.org/titanic-survivor/julia-florence-cavendish.html. Accessed: January 5, 2025.

Cunliffe, Ben. "Julia Florence Cavendish, Little Onn Hall, Church Eaton." Staffordshire Past Track. Staffordshire Past Track. Accessed: January 6, 2026.

### Part 4: Reflection Questions

1. **Database Skills:** SQL helped me narrow down over 800 passengers to a specific group that matched my criteria, such as "females over 70." The biggest challenge I faced was learning the strict syntax; I initially tried using `SORT BY` instead of `ORDER BY`, and I had to learn to use `WHERE` clauses to exclude `None` or null values from my results.

2. **Research Process:** The most difficult part was the "detective work" required to reconcile the data with history, specifically why the database said she was 76 when she was actually 25 during the sinking. I found Encycloedia Brittanica to be the most reliable source because it provided primary documents like death certificates and direct quotes from the survivors.

3. **Historical Connection:** I was surprised that Julia's survival was followed by so much turmoil, like her father going to prison and the strange lawsuit over her home. Learning about her life changed my perspective by showing that the disaster wasn't just a single night of tragedy, but a life-altering event that followed the survivors for decades.

4. **Data vs. Stories:** The database provided the "skeleton" the ticket price of 78 pounds, the cabin C-46, and the fact that she survived. The historical research provided the "heart," explaining her family relationships and her grief. Together, they gave me a full picture of a woman who was a wealthy socialite but also a grieving widow.

5. **Information Literacy:** It is vital to evaluate original sources because data can be incomplete or misleading, as seen with the age discrepancy in my search. In an age of quick answers, checking original records ensures you are getting the true story and not just a summarized or incorrect data point.